python/ray/_private/accelerators/nvidia_gpu.py

```diff
@@ -53,54 +50,17 @@ def get_current_process_visible_accelerator_ids() -> Optional[List[str]]:
 
     @staticmethod
     def get_current_node_num_accelerators() -> int:
-        num_gpus = 0
-        if importlib.util.find_spec("GPUtil"):
-            gpu_list = GPUtil.getGPUs()
-            num_gpus = len(gpu_list)
-        elif sys.platform.startswith("linux"):
-            proc_gpus_path = "/proc/driver/nvidia/gpus"
-            if os.path.isdir(proc_gpus_path):
-                num_gpus = len(os.listdir(proc_gpus_path))
-        elif sys.platform == "win32":
-            props = "AdapterCompatibility"
-            cmdargs = ["WMIC", "PATH", "Win32_VideoController", "GET", props]
-            lines = subprocess.check_output(cmdargs).splitlines()[1:]
-            num_gpus = len([x.rstrip() for x in lines if x.startswith(b"NVIDIA")])
+        num_gpus = gpustat.gpu_count()
```

Comment:
Current `gpustat` is only installed for `ray[default]` so I think we still need the old code that checks "/proc/driver/nvidia/gpus" for minimal installed ray?

Reply:
Would it be acceptable to make gpustat a unconditional dependency for working with GPUs and ray? That code is very fragile.

Reply:
It might be hard since it has external dependencies as well:

```
install_requires = [
    'nvidia-ml-py>=12.535.108',  # see #107, #143, #161
    'psutil>=5.6.0',    # GH-1447
    'blessed>=1.17.1',  # GH-126
    'typing_extensions',
]
```

Reply:
Should we copy the auto detect code that pytorch has `torch.cuda.device_count()`. I think it doesn't depend on GPUtil or gpustat?

Reply:
That goes to [this code](https://github.com/pytorch/pytorch/blob/d5fff7338e1c16a817ea63f07a32cbc58ede5fec/torch/cuda/__init__.py#L698), which eventually [uses ctypes and `libnvidia-ml.so.1`](https://github.com/pytorch/pytorch/blob/d5fff7338e1c16a817ea63f07a32cbc58ede5fec/torch/cuda/__init__.py#L623). How does this work on windows and macos?